_posts/2021-06-09-fed-talk-s01e03-routing.md

```diff
@@ -0,0 +1,513 @@
+This means when you build your App everything is shelled into your projects root `index.html` file made available in the `public` folder. If you create anchor tag links expecting your users to be navigated to another landing URL, it simply will not work as the only `.html` page exported from the build at this stage is the root file.
+This is where the recommended library **React Router** comes into play.
+
+A route is an implementation where we bind the URL into our React App and as React developers we have the power to configure routes in a meaningful way.
+For example we can configure:
+* our home page: `/`,
+* nested child pages: `/product-category/products`,
+* contextual information: `/product-category/products/ABC` -> `/product-category/products/:productId` -> console.log(productId) // "ABC",
+* redirects,
+* fallbacks to things like a "Page not found" page.
+
+---
+
+## Setting Up Routing in React
+
+Before we start implementing we should spend some time upfront to design what our Routes will look like.
+The following questions help me during this phase:
+
+
+**Will your App be publicly available and are you expecting Google (or any other engine) to index your pages?**
+The following topics are worth a read:
+* [Information Architecture](https://www.toptal.com/designers/ia/guide-to-information-architecture){:target="_blank" rel="noopener"}
+* [Why sitemaps matter](https://www.semrush.com/blog/xml-sitemap/){:target="_blank" rel="noopener"}
+
+**Will users copy/paste URLs to deep link into your content?**
+
+**Will users bookmark URLs for future use?**
+
+For the rest of our journey we will build out our App answering the last two questions.
+
+Let's check the current state of our App to see how we can design our Information Architecture.
+
+![Screenshot of the current state of our App](/img/fed-talk/s01e03/component-library-screenshot.png){:width="350"}
+
+There are 3 areas which can be broken down into smaller digestible bits of content: Typographies, Colour Palette, Buttons. Off the bat we can declare 3 routes:
+
+* `/typographies`
+* `/colour-palette`
+* `/buttons`
+
+Take some time to imagine how your App will evolve... I foresee it containing a mixture of information:
+
+* Getting Started (Home page): `/`
+* UI: `/ui/*`
+* Components: `/components/*`
+* Feedback: `/feedback`
+* Page Not found
+
+So, because of this we should change our routes to be:
+
+* `/ui/typographies`
+* `/ui/colour-palette`
+* `/ui/buttons`
+
+Now that we have a clear idea on how our routes can be implemented, lets install the `react-router-dom` library to get started:
+
+```bash
+npm install react-router-dom
+npm install --save-dev @types/react-router-dom
+```
+
+---
+
+## Configuring Routes
+
+It's best to setup Routes at the highest logical level in your App so all the `Router` contextual information can propagate down to your components.
+
+Update your App code with the following:
```

Comment:
It might be better to say:

_Following on from the previous episode, we can update our App code with the following:_

Just to give people context that are reading this series for the first time?